src/Tasks/Common/Resources/Strings.resx

```diff
@@ -887,4 +887,8 @@ You may need to build the project on another operating system or architecture, o
     <value>NETSDK1195: The SDK does not support ahead-of-time compilation. Set the PublishAot property to false.</value>
     <comment>{StrBegin="NETSDK1195: "}</comment>
   </data>
+  <data name="PublishProfileNotPresent" xml:space="preserve">
+    <value>NETSDK1198: A publish profile with the name '{0}' was not found in the project. Set the PublishProfile property to a valid file name</value>
```

Comment:
At the time this warning is created, could we show the list of known or valid profiles? That would make it more actionable for the end user. Other alternatives:

* list only profiles specific to the project being published, never any 'default' profiles
* don't show the list here, and instead do something like https://github.com/dotnet/sdk/issues/26161, pushing validation into that option instead
* do nothing

Reply:
Comments on various options: 
1. We have seen instances where users project may have more than 100+ profiles. So listing all profiles might not be an option.
2. I think having profile an option can still be an orthogonal feature that we can add. Even in this case, they could make a mistake and the warning could still be useful.

Reply:
I do like the option of showing potential profiles.

To circumvent the 100+ issue: We could do something like, show the available profiles that have a 'similar' string to what you entered, such as I gave --profile myPublishProfile, but the real publish profile was `publishprofile`. With that being said, that is probably a p3/p4 issue.

@baronfel Personally I would even like this feature in the SDK, where you type `dotnet built` and it says `did you mean dotnet build?`

Reply:
Since we already have an issue tracking adding support for publishProfile as a top level option - https://github.com/dotnet/sdk/issues/26161, I will mark this conversation as resolved.